target_chains/ethereum/contracts/forge-test/Entropy.t.sol

```diff
@@ -729,4 +731,175 @@ contract EntropyTest is Test, EntropyTestUtils {
         vm.expectRevert();
         random.setProviderUri(newUri);
     }
+
+    function testRequestWithCallback() public {
+        bytes32 protocolRandom = bytes32(uint(42));
+        uint fee = random.getFee(provider1);
+        vm.deal(user1, fee);
+        vm.startPrank(user1);
+        vm.expectEmit(false, false, false, true, address(random));
+        emit RequestedWithCallback(
+            provider1,
+            random.getProviderInfo(provider1).sequenceNumber,
+            protocolRandom
+        );
+        uint64 assignedSequenceNumber = random.requestWithCallback{value: fee}(
+            provider1,
+            protocolRandom
+        );
+
+        assertEq(
+            random.getRequest(provider1, assignedSequenceNumber).requester,
+            user1
+        );
+
+        assertEq(
+            random.getRequest(provider1, assignedSequenceNumber).provider,
+            provider1
+        );
+
+        // requestWithCallback is storing the requests same as
+        // request. If the requester calls the reveal method, it should
+        // succeed.
+        bytes32 randomNumber = random.reveal(
```

Comment:
This test might have caused some confusion. 
Let's be clear about one thing, the requestor:
- whoever calls the request method is the requestor. 
- in case when a contract calls requestWithCallback (it will be a contract almost always), the contract will be the requestor. 
- The user won't be able to call the reveal method. The contract can though. 

If the contract has such intentions of denial of service, they can do so in so many different ways too. We don't need to verify the contract intentions imo.